### PR TITLE
Minor improvements to templates page

### PR DESCRIPTION
**Pull request by dstansby**
- "Domain" > "Community" specific templates
- This makes the first sentence redundant, so remove
- Fix some tool names

### CODE REVIEW

File: docs/pages/templates.md [hunk 1]
```diff
@@ -17,20 +17,20 @@ If you're making a package for a community that already has a template in
 general use (some examples are listed below) we recommend using their template
 instead.
 
-## Domain specific templates
+## Community specific templates
```
Comment (samcunliffe) [inline docs/pages/templates.md]:
```suggestion
## Community-specific templates
```
(Compound adjective)

File: docs/pages/templates.md [hunk 2]
```diff
@@ -17,20 +17,20 @@ If you're making a package for a community that already has a template in
...
 | ------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------ |
 | [Napari](https://github.com/napari/cookiecutter-napari-plugin)     | Cookiecutter template for authoring (npe2-based) napari plugins.                                       |
-| [SciKit Surgery](https://github.com/SciKit-Surgery/PythonTemplate) | Cookiecutter template developed by the Wellcome EPSRC Centre for Interventional and Surgical Sciences. |
-| [SciKit HEP](https://github.com/scikit-hep/cookie)                 | Cookiecutter template developed by SciKit-HEP.                                                         |
+| [SciKit-Surgery](https://github.com/SciKit-Surgery/PythonTemplate) | Cookiecutter template developed by the Wellcome EPSRC Centre for Interventional and Surgical Sciences. |
+| [Scikit-HEP](https://github.com/scikit-hep/cookie)                 | Cookiecutter template developed by SciKit-HEP.                                                         |
```
Comment (samcunliffe) [inline docs/pages/templates.md]:
```suggestion
| [Scientific Python](https://github.com/scientific-python/cookie)                 | Cookiecutter template developed by [SciKit-HEP](https://github.com/scikit-hep) but now adopted by the more general Scientific Python community.                                                         |
```

Which will fail prettier, of course.

**Review by samcunliffe — CHANGES_REQUESTED** [review]
Thanks 👍

File: docs/pages/templates.md (hunk 2 above)
Comment (dstansby) [reply docs/pages/templates.md]: 👍 done

File: docs/pages/templates.md (hunk 1 above)
Comment (dstansby) [reply docs/pages/templates.md]: 👍  done

**Review by samcunliffe — APPROVED** [review]
🫐

**Pull request merged**